doc/explanation/external-builder.md

```diff
@@ -0,0 +1,5 @@
+# External builder
+
+In order to pre-bootstrap LXD, Microk8s snaps to Juju, external VMs are spanwed and then run with
+cloud-init script that installs required components. Then, it is snapshot to work around limitation
+of booting up a fresh image with snaps with state presersverance.
```

Comment:
Nitpick: presersverance --> perseverance 

Clarifying question: What is state perseverance?

Reply:
It preserves the state (bootstrapped state for juju with microk8s and LXD), would there be a better way to phrase itt?

Reply:
Thank you for explaining, I think the current phrasing is fine :)